src/Ocelot.Provider.Polly/OcelotBuilderExtensions.cs

```diff
@@ -1,32 +1,35 @@
-﻿using Microsoft.Extensions.DependencyInjection;
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
 using Ocelot.Configuration;
 using Ocelot.DependencyInjection;
 using Ocelot.Errors;
 using Ocelot.Logging;
+using Ocelot.Provider.Polly.Interfaces;
 using Ocelot.Requester;
 using Polly.CircuitBreaker;
 using Polly.Timeout;
-
-namespace Ocelot.Provider.Polly
-{
-    public static class OcelotBuilderExtensions
-    {
-        public static IOcelotBuilder AddPolly(this IOcelotBuilder builder)
-        {
-            var errorMapping = new Dictionary<Type, Func<Exception, Error>>
-            {
-                {typeof(TaskCanceledException), e => new RequestTimedOutError(e)},
-                {typeof(TimeoutRejectedException), e => new RequestTimedOutError(e)},
-                {typeof(BrokenCircuitException), e => new RequestTimedOutError(e)},
-            };
-
-            builder.Services
-                .AddSingleton(errorMapping)
-                .AddSingleton<QosDelegatingHandlerDelegate>(GetDelegatingHandler);
-            return builder;
-        }
 
-        private static DelegatingHandler GetDelegatingHandler(DownstreamRoute route, IOcelotLoggerFactory logger)
-            => new PollyCircuitBreakingDelegatingHandler(new PollyQoSProvider(route, logger), logger);
-    }
+namespace Ocelot.Provider.Polly;
+
+public static class OcelotBuilderExtensions
+{
+    public static IOcelotBuilder AddPolly(this IOcelotBuilder builder)
+    {
+        var errorMapping = new Dictionary<Type, Func<Exception, Error>>
+        {
+            { typeof(TaskCanceledException), e => new RequestTimedOutError(e) },
+            { typeof(TimeoutRejectedException), e => new RequestTimedOutError(e) },
+            { typeof(BrokenCircuitException), e => new RequestTimedOutError(e) },
+            { typeof(BrokenCircuitException<HttpResponseMessage>), e => new RequestTimedOutError(e) },
+        };
+
+        builder.Services
+            .AddSingleton(errorMapping)
+            .AddSingleton<IPollyQoSProvider<HttpResponseMessage>, PollyQoSProvider>()
+            .AddSingleton<QosDelegatingHandlerDelegate>(GetDelegatingHandler);
```

Comment:
Not an issue. But could we rename it removing suffix `Delegate` ?
According to Microsoft .NET conventions, [EventHandler](https://learn.microsoft.com/en-us/dotnet/api/system.eventhandler?view=net-7.0) or other `Handler` types are delegates.
So, there is no need to add this suffix, it is redundant.

Reply:
Not changed! The [link](https://github.com/ThreeMammals/Ocelot/pull/1753/files#diff-586799ba26fd065bccd320cb27577637fa83b08a9ec758a28c792294115086b0R7)
It returns `DelegatingHandler`!